### DISCUSSION TITLE
mini.pick - pass highlighted text to grep_live and buf_lines

### DISCUSSION THREAD

**Discussion by beaumccartney** [nvim-mini/mini.nvim#753] [General]
Is it possible to pass the currently highlighted text as the search to grep_live and buf_lines? I'm thinking like what `*` does in vmode but put everything in a picker.

**Comment by echasnovski** [comment]
Not for these pickers, no.

There is a way for `builtin.grep()` though through `opts.pattern`. For example, to search for a word under cursor (as `*` does), execute `:Pick grep pattern="<cword>"`. Making it to a mapping with `'<Cmd>Pick grep pattern="<cword>"<CR>'` as right hand side is more usable.

**Reply by beaumccartney** [reply]
cword only does the first word highlighted. If the highlighted text contains spaces (or other delimiters?) <cword> only expands to the first word.

e.g. `:echo expand("<cword>") with `a b c` highlighted only prints `a`

**Reply by echasnovski** [reply]
If you **specifically** want to use the visual selection, then I am afraid there is no easy solution here. Eventually, it can be done by manually extracting text in visual selection and supplying it as `opts.pattern`.
